amqp/json_schemas/channel.json

```diff
@@ -0,0 +1,68 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://asyncapi.com/amqp/bindings/channel.json",
+  "title": "Channel Schema",
+  "type": "object",
+  "additionalProperties": false,
+  "patternProperties": {
+    "^x-[\\w\\d\\.\\-\\_]+$": {
+      "$ref": "https://github.com/asyncapi/asyncapi-node/blob/master/schemas/2.0.0.json#/definitions/specificationExtension"
+    }
+  },
+  "properties": {
```

Comment:
in spec for amqp you can find

> Defines what type of channel is it. Can be either queue or routingKey (default).
> When is=routingKey, this object defines the exchange properties.
> When is=queue, this object defines the queue properties.

So you can either have `routingKey` or `queue`, but not both. Yes, first of all looks like example is wrong.
You need to therefore modify the json schema a bit as you need to use `oneOf` to describe that only one of the objects is allowes, either `exchange` or `queue` depending on the `is` property.

It is another level of JSON Schema 😄 . Feel free to let me know if you need some help here. I worked on something similar recently [here](https://gist.github.com/derberg/4e419d6ff5870c7c3f5f443e8bd30535#file-asyncapi-websocket-kraken-yml-L259-L313) and [here](https://github.com/asyncapi/spec/pull/533/files#diff-8cbe9bd49981ca789c9540337203e60bc894cb6a1ce98315a4b5c30d5199d833R127-R176)